### ISSUE TITLE
500 instead of 404 when getting a blob that doesn't exist with public-read bucket

### ISSUE THREAD

**Issue by kahing** [andrewgaul/s3proxy#182]
```
W 01-18 00:57:08.679 S3Proxy-18 o.e.jetty.server.HttpChannel:396 |::] http://127.0.0.1:8080/nrcxt0gq4l7b3kpm/test
java.lang.RuntimeException: java.nio.file.NoSuchFileException: /tmp/s3proxy/nrcxt0gq4l7b3kpm/test
        at com.google.common.base.Throwables.propagate(Throwables.java:160)
        at org.jclouds.filesystem.strategy.internal.FilesystemStorageStrategyImpl.getBlobAccess(FilesystemStorageStrategyImpl.java:574)
        at org.jclouds.blobstore.config.LocalBlobStore.getBlobAccess(LocalBlobStore.java:403)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:37)
        at com.sun.proxy.$Proxy39.getBlobAccess(Unknown Source)
        at org.gaul.s3proxy.S3ProxyHandler.doHandleAnonymous(S3ProxyHandler.java:662)
        at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:360)
        at org.gaul.s3proxy.S3ProxyHandler.handle(S3ProxyHandler.java:260)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:499)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.nio.file.NoSuchFileException: /tmp/s3proxy/nrcxt0gq4l7b3kpm/test
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileAttributeViews$Posix.readAttributes(UnixFileAttributeViews.java:218)
        at sun.nio.fs.UnixFileAttributeViews$Posix.readAttributes(UnixFileAttributeViews.java:131)
        at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
        at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
        at java.nio.file.Files.readAttributes(Files.java:1737)
        at java.nio.file.Files.getPosixFilePermissions(Files.java:2004)
        at org.jclouds.filesystem.strategy.internal.FilesystemStorageStrategyImpl.getBlobAccess(FilesystemStorageStrategyImpl.java:572)
        ... 18 common frames omitted
```

**Comment by andrewgaul** [comment]
Fixed in jclouds 2.0.1-SNAPSHOT although not bumping the dependency in S3Proxy yet.  You can use the transient blobstore as a workaround for now.

**Comment by andrewgaul** [comment]
Fixed in 62b7c53eae9cf7783760afc10c171233aa2804b4.
